mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -534,8 +577,15 @@ class Word2VecModel private[spark] (
     // Need not divide with the norm of the given vector since it is constant.
     val cosVec = cosineVec.map(_.toDouble)
     var ind = 0
+    var vecNorm = 1f
+    if (norm) {
```

Comment:
OK. Let me pull back the changes about normalization then. we can use the other pull request for follow up of normalization. 
As @srowen mentioned, normalization can be done after getting top K in the application as well.